pallets/scoring/src/lib.rs

```diff
@@ -0,0 +1,359 @@
+		// Invalid rating value - should be between 1 and 50
+		InvalidRatingValue,
+		// Exceeds the maximum allowed entries in a single transaction
+		TooManyJournalEntries,
+		// Invalid entity signature
+		InvalidEntitySignature,
+		//Stream digest is not unique
+		DigestAlreadyAnchored,
+		//Count entry Greater than storage count
+		CountGreaterThanStorage,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		///
+		/// Create a new rating identifier and associates it with its
+		/// controller. The controller (issuer) is the owner of the identifier.
+		///
+		/// * origin: the identity of the Transaction Author. Transaction author
+		///   pays the transaction fees
+		/// * tx_journal: the incoming rating entry.
+		/// * `authorization`: The authorization ID of the delegate who is
+		///   allowed to perform this action.
+		#[pallet::call_index(0)]
+		#[pallet::weight({0})]
+		pub fn entries(
+			origin: OriginFor<T>,
+			journal: RatingInputOf<T>,
+			authorization: AuthorizationIdOf,
+		) -> DispatchResult {
+			let author = <T as Config>::EnsureOrigin::ensure_origin(origin)?.subject();
+
+			let registry_id =
+				pallet_registry::Pallet::<T>::is_a_delegate(&authorization, author.clone(), None)
+					.map_err(<pallet_registry::Error<T>>::from)?;
+
+			ensure!(
+				(journal.entry.rating >= T::MinScoreValue::get() &&
+					journal.entry.rating <= T::MaxScoreValue::get()),
+				Error::<T>::InvalidRatingValue
+			);
+			ensure!(
+				!<JournalHashes<T>>::contains_key(&journal.digest),
+				Error::<T>::DigestAlreadyAnchored
+			);
+
+			let identifier = Ss58Identifier::to_scoring_id(&(&journal.digest).encode()[..])
+				.map_err(|_| Error::<T>::InvalidIdentifierLength)?;
+
+			ensure!(
+				!<TidEntries<T>>::contains_key(&journal.entry.tid, &journal.entry.rating_type),
+				Error::<T>::TransactionAlreadyRated
+			);
+
+			let block_number = frame_system::Pallet::<T>::block_number();
+
+			Self::aggregate_score(&journal.entry);
+
+			<Journal<T>>::insert(
+				&identifier,
+				&journal.entry.rating_type,
+				RatingEntryOf::<T> {
+					entry: journal.entry.clone(),
+					digest: journal.digest,
+					created_at: block_number,
+					registry: registry_id,
+					creator: author.clone(),
+				},
+			);
+			<JournalHashes<T>>::insert(&journal.digest, ());
+			<TidEntries<T>>::insert(
+				&journal.entry.tid,
+				&journal.entry.rating_type,
+				&journal.entry.entity,
+			);
+
+			Self::deposit_event(Event::JournalEntry {
+				identifier,
+				entity: journal.entry.entity.clone(),
+				author: author.clone(),
+			});
+
+			Ok(())
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	pub fn aggregate_score(entry: &RatingDetailsOf<T>) -> Result<(), pallet::Error<T>> {
+		if let Some(aggregate) = <Scores<T>>::get(&entry.entity, &entry.rating_type) {
+			let total_count = aggregate.count + entry.count;
+
+			let sum = match entry.entry_type {
+				RatingEntryType::Credit =>
+					Self::sum_average(aggregate.rating + entry.rating, total_count),
```

Comment:
Don't do average. We only need to keep the aggregate of counts and scores. The average function will be at the SDK level as the value received at the chain is base-10 encoded.

Reply:
also, use saturating_add and saturating_sub instead of +, - operators. (https://doc.rust-lang.org/std/intrinsics/fn.saturating_add.html#)

Reply:
> Don't do average. We only need to keep the aggregate of counts and scores. The average function will be at the SDK level as the value received at the chain is base-10 encoded.

@adi-a11y   - FYI, the average calculation has to be performed in the SDK